modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java

```diff
@@ -138,14 +141,25 @@ private Map<String, Object> _getObjectEntryProperties(
 	}
 
 	private Object _getRelatedObjectValue(
-		ObjectEntry objectEntry, APIApplication.Property property,
-		List<String> relationshipsNames) {
+			ObjectEntry objectEntry, APIApplication.Property property,
+			List<String> relationshipsNames)
+		throws Exception {
 
 		if (relationshipsNames.isEmpty()) {
 			Map<String, Object> objectEntryProperties =
 				objectEntry.getProperties();
 
-			return objectEntryProperties.get(property.getSourceFieldName());
+			Object object = objectEntryProperties.get(
```

Comment:
I would purpose to implement in REST Builder a method in the DTO (in this case, in ObjectEntry) to get a property based on its name. As a property is set with `x-json-map: true`, that means Jackson will include any unknown property into the map, so we have the power in REST Builder to know if a property is given by its field or by the map.

Something like:

```
public Object getProperty(String propertyName) {
	if (Objects.equals(propertyName, "actions")) {
		return actions;
	} else if (Objects.equals(propertyName, "auditEvents")) {
		return auditEvents;
	} else if (Objects.equals(propertyName, "creator")) {
		return creator;
		
	[...]
		
	} else if (Objects.equals(propertyName, "externalReferenceCode")) {
		return externalReferenceCode;
	} else {
		return properties.get(propertyName);
	}
}
```


What do you think @4lejandrito ??